lib/dao.js

```diff
@@ -1846,10 +1846,40 @@ DataAccessObject.find = function find(query, options, cb) {
           }
         }
 
-        if (connector.all.length === 4) {
-          connector.all(self.modelName, {}, options, geoCallback);
+        function geoCallbackWithNotify(err, data) {
+          if (err) return cb(err);
+
+          async.map(data, function(item, next) {
+            var context = {
+              Model: self,
+              data: item,
+              isNewInstance: false,
+              hookState: hookState,
+              options: options,
+            };
+
+            self.notifyObserversOf('loaded', context, function(err) {
+              if (err) return next(err);
+              next(null, context.data);
+            });
+          }, function(err, results) {
+            if (err) return cb(err);
+            geoCallbackWithoutNotify(null, results);
+          });
+        }
+
+        if (options.notify === false) {
```

Comment:
@bajtos Thanks for this great suggestion; I love it 👍  however please note the right code would be the following:
```
        var geoCallback = options.notify === false ? geoCallbackWithoutNotify : geoCallbackWithNotify;
        if (connector.all.length === 4) {
          connector.all(self.modelName, {}, options, geoCallback);
        } else {
          connector.all(self.modelName, {}, geoCallback);
        }
```

The reason is if options.notify is `undefined` we want to go for `geoCallbackWithNotify`, so we need to check it's value with `false`.

Thanks for this great suggestion which makes the code cleaner :-)